ServiceManager/GUI/main_window.py

```diff
@@ -0,0 +1,629 @@
+                item = self.config_table.item(rowIndex, column_of_interest)
+                if value_of_interest in item.text():
+                    self.config_table.setRowHidden(rowIndex, False)
+                else:
+                    self.config_table.setRowHidden(rowIndex, True)
+
+    def clear_filters(self):
+        self.filter_columns_cb.setCurrentIndex(0)
+        self.filter_bar.clear()
+        self.refresh_config()
+
+    def new_config_btn_clicked(self):
+        """ Open the Config Entry dialog window. If database is not connected, display error message. """
+        if not DBUtils.is_connected():
+            QMessageBox.critical(self, 'Database connection required',
+                                       'Database connection is required to edit the PV configuration.',
+                                       QMessageBox.Ok)
+            return
+
+        if self.config_entry_w is None:
+            self.config_entry_w = UIConfigEntryDialog()
+            self.config_entry_w.config_updated.connect(self.refresh_config)
+        self.config_entry_w.show()
+        self.config_entry_w.activateWindow()
+
+    def edit_config_btn_clicked(self):
+        """ On Edit, open Config Entry dialog window as normal with object selected and PV config loaded. """
+        if not DBUtils.is_connected():
+            QMessageBox.critical(self, 'Database connection required',
+                                       'Database connection is required to edit the PV configuration.',
+                                       QMessageBox.Ok)
+            return
+        if self.config_entry_w is None:
+            self.config_entry_w = UIConfigEntryDialog()
+            self.config_entry_w.config_updated.connect(self.refresh_config)
+        self.config_entry_w.show()
+
+        selected_row_items = self.config_table.selectedItems()
+        if selected_row_items:
+            selected_object_name = selected_row_items[1].text()
+            self.config_entry_w.edit_object_config(obj_name=selected_object_name)
+
+        self.config_entry_w.activateWindow()
+
+    def delete_config_btn_clicked(self):
+        """
+        Display message box with list of selected PV config entries, with Delete & Cancel options.
+        On Delete, the entries will be deleted from the PV config, and table will be refreshed.
+        """
+        selected = self.config_table.selectedIndexes()
+        cols_count = self.config_table.columnCount()
+        selected_rows = set()
+        for index in selected[1::cols_count]:
+            selected_rows.add(index.row())
+
+        if not selected_rows:
+            return
+
+        obj_ids = []  # ids of objects whose config is to be removed
+        obj_list = []
+        for row_no in selected_rows:
+            id_ = self.config_table.item(row_no, 0)
+            obj_ids.append(id_.text())
+            name_ = self.config_table.item(row_no, 1)
+            obj_list.append(f'(ID: {id_.text()}) {name_.text()}')
```

Comment:
**should**: You should pull out the which column index is the name/id into a constant. It's used in a few other places too